.github/workflows/analysis.yml

```diff
@@ -0,0 +1,40 @@
+name: Static Analysis
+on: [push, pull_request]
+permissions:
+  contents: read
+
+jobs:
+  static_analysis:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - uses: actions/setup-go@v5
+      with:
+        go-version: 'stable'
+
+    - name: Install analysis tools
+      run: |
+        go install golang.org/x/tools/cmd/goimports@latest
+        go install github.com/fzipp/gocyclo/cmd/gocyclo@latest
+        go install honnef.co/go/tools/cmd/staticcheck@latest
+
+    - name: Prepare
+      run: rm -rf v3.0/ v3.1/ v3.2/
+
+    - name: Get dependencies
+      run: sudo apt-get update && sudo apt-get install gcc libgl1-mesa-dev libegl1-mesa-dev libgles2-mesa-dev libx11-dev xorg-dev
+
+    - name: Vet
+      run: go vet ./...
+
+    - name: Goimports
+      run: test -z "$(goimports -e -d . | tee /dev/stderr)"
+
+    - name: Gocyclo
+      run: gocyclo -over 15 .
```

Comment:
Not sure how helpful this is. If you prefer to add it, I'm okay with it, but if it starts to produce reports that we deem to be false positives, let's drop it then.

Reply:
I double the complexity budget so we aren't too strict. We have been using this in fyne since basiclaly forever and it is a good way of keeping functions short and readable